### PR TITLE
ellipticity distribution of Ryden (2004)

### PR DESCRIPTION
**Pull request by ntessore**
## Description

Adds the ellipticity distribution following [Ryden (2004)](https://ui.adsabs.harvard.edu/abs/2004ApJ...601..214R).

## Checklist
- [x] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/master/CONTRIBUTING.md)
- [x] Write unit tests
- [x] Write documentation strings
- [x] Assign someone from your working team to review this pull request
- [x] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

File: skypy/galaxy/ellipticity.py [hunk 1]
```diff
@@ -95,3 +97,83 @@ def fit(self, data, *args, **kwargs):
...
+    size = np.broadcast(gam, eps).shape
+
+    # draw random viewing angle (theta, phi)
+    cos2_theta = stats.uniform.rvs(loc=-1.0, scale=2.0, size=size)**2
+    cos2_phi = np.cos(stats.uniform.rvs(scale=2*np.pi, size=size))**2
+    sin2_theta = 1 - cos2_theta
+    sin2_phi = 1 - cos2_phi
+
+    # compute Binney's ABC -- eq.s (13)-(15)
+    A = (1 - eps*(2-eps)*sin2_phi)*cos2_theta + gam**2*sin2_theta
```
[the remark this answers is not in the document]
Comment (ntessore) [reply skypy/galaxy/ellipticity.py:171]: That is true for input arrays where you don't know if they support the operation. Here I have sampled `gam` myself and I know `gam**2` works.

**Comment by ntessore** [comment]
> Not knowing the theory behind it, it looks fine for me. But is there a way to maybe test the output against some theory values?

Not really, except checking by eye that the parameters from the paper reproduce the figures from the paper, which I did check.

**Comment by philipp128** [comment]
> > Not knowing the theory behind it, it looks fine for me. But is there a way to maybe test the output against some theory values?
> 
> Not really, except checking by eye that the parameters from the paper reproduce the figures from the paper, which I did check.

Ok, in case calculating the power with `**` or `np.power` is not important I would approve from my site.

**Comment by ntessore** [comment]
> Looks good, I can follow the maths from the reference. I think we need to have some tests for the sampled values even though the distribution is so complex. My suggestions:
> 
> * Check that the sampled values are all between 0 and 1 for arbitrary parameters
> * I think if mu_gamma -> 1, mu -> -inf and sigma, sigma_gamma -> 0 then all galaxies are spherical (A=B=C) so the sampled values should all be roughly zero?

Those are always true, but might guard against some mess-up in the future? I'll add them.

**Pull request merged**